### ISSUE TITLE
Remove dat.ErrNotFound

### ISSUE THREAD

**Issue by sbowman** [mgutz/dat#37]
Half the query functions in dat use the standard "sql.ErrNoRows" and half use "dat.ErrNotFound".  Could you please just standardize on "sql.ErrNoRows" and drop "dat.ErrNotFound", so we can stop having to guess which is which or testing for both all the time?

**Comment by mgutz** [comment]
I agree with this. `dat` used to work with another postgres driver (pgx) and I was being too clever in my abstractions. The next version will remove it. I can't remove it from the current version as it will break API compatibility.
